.github/workflows/release.yaml

```
@@ -0,0 +1,74 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Ensure the branch is protected
+        run: |
+          if [[ "${{ github.ref_protected }}" == "false" ]]; then
+            echo "::error::The branch ${{ github.ref }} is not protected"
+            exit 1
+          fi
+      - name: Ensure tag has not been released
+        run: |
+          TAG_EXISTS=$(git tag -l "${{ inputs.tag }}")
+          if [ -n "$TAG_EXISTS" ]; then
+            echo "::error::The tag ${{ inputs.tag }} has already been released"
+            exit 1
+          fi
+      - name: Ensure version is properly set
+        run: |
+           PKG_VERSION="$(yarn -s run get-version)"
+           if ! [ "$PKG_VERSION" = "$VERSION" ]; then
+             echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag $VERSION must match the version $PKG_VERSION specified in package.json"
+             exit 1
+           fi
+        env:
+          VERSION: ${{ inputs.tag }}
+      - name: Setup node with GitHub Packages
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+          node-version: '16'
+          registry-url: https://npm.pkg.github.com
+      - run: yarn install --frozen-lockfile
+      - name: Publish to GitHub Packages
+        run: npm publish --provenance
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup node with npmjs.org
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16'
+          registry-url: https://registry.npmjs.org
+      - name: Publish to npmjs.org
+        run: npm publish --provenance
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ inputs.tag }}
+          name: Release ${{ inputs.tag }}
+          target_commitish: ${{ github.sha }}
```

Can we use `append_body` to include the url to the release version in each registry as well in the body for easy access

```suggestion
          target_commitish: ${{ github.sha }}
          append_body: |
            Github Packages: https://github.com/$GITHUP_REPO/pkgs/npm/${{ pkg_name }}/${{ pkg_version here or random number ? }}
            npmjs: https://www.npmjs.com/package/${{ pkg_name }}/v/${{ pkg_name }}
        env:
          pkg_name: $(sed -E '/"name":/!d; s/.*"name": "(.*?)",/\1/g' package.json)
          pk_version: $(yarn -s get-version)
```

---

done